crates/uv-interpreter/src/environment.rs

```diff
@@ -12,7 +13,10 @@ use crate::{find_default_python, find_requested_python, Error, Interpreter, Targ
 
 /// A Python environment, consisting of a Python [`Interpreter`] and its associated paths.
 #[derive(Debug, Clone)]
-pub struct PythonEnvironment {
+pub struct PythonEnvironment(Arc<SharedPythonEnvironment>);
+
+#[derive(Debug, Clone)]
+struct SharedPythonEnvironment {
```

Comment:
Just for my knowledge, is this the typical naming scheme for this pattern?

Reply:
I usually use `Foo` and `FooInner` personally. I don't think I've seen `SharedFoo` much? I like adding a suffix personally. So I'd prefer `FooShared` (or whatever). But I don't have a strong opinion.

Reply:
Cool thanks! `Inner` makes a bit more sense to me.

Reply:
I sort of avoid `Inner` because it feels like a catch-all naming convention. A suffix seems slightly better for readability.